ADAL/src/request/ADWebRequest+NSURLSession.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@class ADWebRequest;
+@class ADWebResponse;
+
+typedef enum
+{
+    ADWebRequestGet,
+    ADWebRequestPost,
+} ADWebRequestMethodType;
+
+@interface ADWebRequest : NSObject
+{
+    NSURL * _requestURL;
+    NSMutableDictionary * _requestHeaders;
+    NSData * _requestData;
+    
+    NSHTTPURLResponse * _response;
+    NSMutableData * _responseData;
+    
+    NSUUID * _correlationId;
+    
+    NSUInteger _timeout;
+    
+    NSOperationQueue * _operationQueue;
+    
+    BOOL _isGetRequest;
+    
+    void (^_completionHandler)(NSError *, ADWebResponse *);
+}
+
+@property (strong, readonly, nonatomic)     NSURL               *URL;
+@property (copy, nonatomic)                 NSMutableDictionary *headers;
```

Comment:
The ADWebRequest+NSURLSession is a "copy" of ADWebRequest source code except the usage of NSURLSession, especially the declaration in the header files are pretty much the same. If this would be a problem, should we surface it back to main repository?